app/jobs/appointment_reminders_job.rb

```diff
@@ -0,0 +1,14 @@
+class AppointmentRemindersJob < ApplicationJob
+  queue_as :default
+
+  def perform
+    Appointment.pending.within_reminder_window.each do |appointment|
+      sent_in_current_period = ReminderActivity
+                               .where(appointment: appointment)
+                               .where(created_at: 12.hours.ago..12.hours.from_now)
```

Comment:
You could collapse the wheres and the any check into a single `exists?` with the conditions.